internal/services/pubkey_service.go

```diff
@@ -99,6 +117,48 @@ func DeletePubkey(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	pubkeyResource, err := dao.GetPubkeyResourceDao(r.Context())
+	if err != nil {
+		renderError(w, r, payloads.NewDAOError(r.Context(), "delete pubkey", err))
+		return
+	}
+
+	resources, err := pubkeyResource.ListByPubkeyId(r.Context(), id)
+	if err != nil {
+		renderError(w, r, payloads.NewDAOError(r.Context(), "delete pubkey", err))
+		return
+	}
+
+	for _, res := range resources {
+		if res.Handle != "" {
+			arn, err := sourcesClient.GetArn(r.Context(), res.SourceID)
+			if err != nil {
+				if errors.Is(err, sources.ApplicationNotFoundErr) {
+					renderError(w, r, payloads.ClientError(r.Context(), "Sources", "can't fetch arn from sources: application not found", err, 404))
+					return
+				}
+				if errors.Is(err, sources.AuthenticationForSourcesNotFoundErr) {
+					renderError(w, r, payloads.ClientError(r.Context(), "Sources", "can't fetch arn from sources: authentication not found", err, 404))
+					return
+				}
+				renderError(w, r, payloads.ClientError(r.Context(), "Sources", "can't fetch arn from sources", err, 500))
+				return
+			}
+			crd, err := stsClient.AssumeRole(arn)
+			if err != nil {
+				renderError(w, r, payloads.ClientError(r.Context(), "Sources", "cannot assume role: %w", err, 500))
```

Comment:
The ClientError is reserved only for errors coming from HTTP clients (sources, image builder). Of all other errors, find an suitable error in the payloads package, or create a new one.

```suggestion
				renderError(w, r, payloads.NewAWSError(r.Context(), "assume role sts", err))
```